### PR TITLE
CMVNPLG-20: Add documentation

### PR DESCRIPTION
**Pull request by Niranjan-K**
Moved documentation from WSO2 Carbon 5.1.0 to Github and added documention for P2 touchpoints

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,16 +2,171 @@
 
 This repository provides supplementary maven plugins for WSO2 carbon developers.
 
-carbon-feature-plugin
-=====================
+# Using the Carbon Feature Plugin
 
-Carbon-Feature-Plugin (formerly known as carbon-p2-plugin) is a maven plugin developed within the WSO2 and maintained by WSO2 carbon team.
```
Comment (nilminiwso2) [inline README.md:8]: *a maven plugin developed within WSO2 and maintained by the WSO2 Carbon team.

File: README.md [hunk 2]
```diff
@@ -2,16 +2,171 @@
...
+The Carbon feature plugin was formerly known as the Carbon P2 plugin. It has gone through several development cycles in the previous WSO2 Carbon versions. However, the plugin introduced from WSO2 Carbon 5 (Carbon 5.x.x platform) is redesigned to update the Maven APIs, to redefine the configurations by removing any existing inconsistencies, to make its architecture solid, and also to improve the understandability and maintainability of its coding standards.
 
-##How to Contribute
```
Comment (nilminiwso2) [inline README.md:10]: Space required after ## ?

File: README.md [hunk 3]
```diff
@@ -2,16 +2,171 @@
...
-* Please report issues at [WSO2 JIRA](https://wso2.org/jira/browse/CMVNPLG).
+## Introduction
+
+The WSO2 Carbon feature plugin is a Maven plugin, which is used within the WSO2 Carbon platform. The Maven goals that are achieved through this plugin are explained in the following table:
```
Comment (nilminiwso2) [inline README.md]: *are explained below.

File: README.md [hunk 4]
```diff
@@ -2,16 +2,171 @@
...
+
+            <feature>
+               <id>org.wso2.ciphertool.feature.group</id>
+               <version>4.4.0</version>
+            </feature>
+        </features>
+ 
+ > Add the ID of the feature being installed as the value of the `<ID>` property, and add the version of the feature being installed as the value of the `<version>` property.
+ 
+* `deleteOldRuntimeFiles`: Whether to delete old `*.profile` folders located in the `<CARBON_HOME>/repository/components/p2/org.eclipse.equinox.p2.engine/profileRegistry/` directory. The default value is set to true.
```
Comment (nilminiwso2) [inline README.md]: *Specifies whether to delete old `*.profile` folders located in the ..... directory.

File: README.md [hunk 5]
```diff
@@ -2,16 +2,171 @@
...
+          </build>
+	  
+You can modify the above file to add the configurations of the plugin by adding the following parameters within the `<configuration>` element of it. 
+
+* `destination`: Points to the `<CARBON_HOME>/repository/components/` directory.
+
+ MANDATORY property.
+ Example: `<destination>/home/Carbon/wso2carbon-4.4.0/repository/components</destination>`
+ 
+* `runtime`: Which runtime in the destination from where you need to uninstall the features.
```
Comment (nilminiwso2) [inline README.md]: *The runtime required (in the destination from where you need to uninstall the features).

File: README.md [hunk 6]
```diff
@@ -2,16 +2,171 @@
...
+             <feature>
+                <id>org.wso2.ciphertool.feature.group</id>
+                <version>4.4.0</version>
+             </feature>
+        </features>
+        
+
+## P2 Touchpoint
+### Equinox P2 Touchpoint
+Features can have configuration files which need to be copied to a particular location when install the feature. This can be done by specifying the location in the p2.inf in the feature. For example:
```
Comment (nilminiwso2) [inline README.md]: Features can have configuration files that need to be copied to a particular location when installing the feature.

File: README.md [hunk 7]
```diff
@@ -2,16 +2,171 @@
...
+Example:
+instructions.configure = \
+org.eclipse.equinox.p2.touchpoint.natives.copy(source:${installFolder}/../lib/features/org.wso2.carbon.kernel_${feature.version}/launcher/,target:${installFolder}/../../bin/bootstrap/,overwrite:true);\
+
+All equinox p2 touchpoint are available at [Provisioning Actions and Touchpoints](http://help.eclipse.org/neon/index.jsp?topic=/org.eclipse.platform.doc.isv/guide/p2_actions_touchpoints.html)
+
+### Copy to specific runtime
+In C5, there are runtimes which are specific to each product and the feature developer will not know this. In order to support dynamically copying files to the runtime location during feature installation, WSO2 has introduced a custom touchpoint.
```
Comment (nilminiwso2) [inline README.md]: *In Carbon 5, there are product-specific runtimes, which the developer will not know about.

File: README.md [hunk 8]
```diff
@@ -2,16 +2,171 @@
...
+Example:
+instructions.configure = \
+org.eclipse.equinox.p2.touchpoint.natives.copy(source:${installFolder}/../lib/features/org.wso2.carbon.kernel_${feature.version}/launcher/,target:${installFolder}/../../bin/bootstrap/,overwrite:true);\
+
+All equinox p2 touchpoint are available at [Provisioning Actions and Touchpoints](http://help.eclipse.org/neon/index.jsp?topic=/org.eclipse.platform.doc.isv/guide/p2_actions_touchpoints.html)
+
+### Copy to specific runtime
+In C5, there are runtimes which are specific to each product and the feature developer will not know this. In order to support dynamically copying files to the runtime location during feature installation, WSO2 has introduced a custom touchpoint. 
+
+In the p2.inf, the following should be added in-order to copy a particular file to a runtime location:
```
Comment (nilminiwso2) [inline README.md]: *The following should be added to the p2.inf in order to copy a particular file to a runtime location:

File: README.md [hunk 9]
```diff
@@ -2,16 +2,171 @@
...
+
+> metaRequirements.0.namespace = org.eclipse.equinox.p2.iu
+> metaRequirements.0.name = org.wso2.carbon.p2.touchpoint
+> 
+> instructions.configure = \
+> org.wso2.carbon.p2.touchpoint.copy(source:${installFolder}/../lib/features/org.wso2.carbon.touchpoint.sample_${feature.version}/bin/,target:${installFolder}/../\<runtime\>/bin/, overwrite:true);\
+> org.wso2.carbon.p2.touchpoint.copy(source:${installFolder}/../lib/features/org.wso2.carbon.touchpoint.sample_${feature.version}/conf/osgi/launch.properties,target:${installFolder}/../\<runtime\>/conf/osgi/launch.properties, overwrite:true);\
+
+where `<runtime>` which is replaced with the runtime name at the feature installation
```
Comment (nilminiwso2) [inline README.md]: ??

File: README.md [hunk 10]
```diff
@@ -2,16 +2,171 @@
...
+
+This custom touchpoint should be available in the p2-repo. Therefore it should be added to `generate-repo` goal as below in the product generation:
```
Comment (nilminiwso2) [inline README.md:138]: *Therefore, it should be added to the `generate-repo` goal as shown below in the product generation:

File: README.md (hunk 9 above)
Comment (Niranjan-K) [reply README.md]: This is an explanation for the value `runtime` which is used in the above code sample

File: README.md (hunk 1 above)
Comment (Niranjan-K) [reply README.md:8]: This is removed. The green one is the newly added things

File: README.md (hunk 2 above)
Comment (Niranjan-K) [reply README.md:10]: This is removed content

**Comment by Niranjan-K** [comment]
Closing this PR as this is fixed in #52

**Pull request closed without merging**